### PR TITLE
Improve linter configuration

### PR DESCRIPTION
**Pull request by bbenoist**
# Feature

## Work performed

* Remove some react/no-array-index-key eslint warnings 
* Remove func-names eslint warnings and make it an error 
* Don't make eslint warn about console and alert usage in Storybook files
* Remove useless console.log in classification-tag.tsx
* Add minimal VS Code extension recommendations
* Lint Markdown files with markdownlint
* In the changelog, add an extra new line at the end of each release
* Re-enable the ESLint react-hooks plugin and fix the errors

## Results

No new feature or bugfix. This PR only includes developer experience improvements.

## Problems encountered

I cannot remove every `no-console` eslint warning without taking an opinionated decision:
Either:
* remove the eslint `no-console` rule
* disable it locally with `// eslint-disable-next-line no-console` 
* replace the `console.log` usages by an alternative.

Also, a lot of react hooks dependencies warnings have appeared while enabling the ESLint react-hooks plugin.  
While they are legitimate cases (e.g. with `useEffect1), I think that I'd be better if we add the missing dependencies where appropriate and ignore the remaining warnings with appropriate justification comments.

## Caveats

N/A

## Resolved issues

N/A

## Newly raised issues

N/A

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,6 +1,11 @@
 {
-  "extends": ["airbnb-typescript", "plugin:prettier/recommended"],
+  "extends": [
+    "airbnb-typescript",
```
Comment (RaynalHugo) [inline .eslintrc:3]: Small suggestions: 

While we are at it, what do you think of adding [next.js eslint plugin](https://nextjs.org/docs/basic-features/eslint)?

Every time we start the web app, it displays a warning saying we should probably use this.

(We can do it in another PR if you prefer)

File: .markdownlintignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+/CHANGELOG.md
+node_modules
+python/iog
```
Comment (RaynalHugo) [inline .markdownlintignore:3]: Why do we want to ignore python/iog ?

File: strapi/README.md [hunk 3]
```diff
@@ -6,7 +6,7 @@ This uses NPM, for compatibility reasons (See https://github.com/strapi/strapi/i
 
 To run it:
 
-```
+```text
```
Comment (RaynalHugo) [inline strapi/README.md:9]:
```suggestion
```sh
```

File: strapi/README.md [hunk 4]
```diff
@@ -20,21 +20,21 @@ Follow this https://strapi.io/documentation/developer-docs/latest/setup-deployme
 
 Then once setup is done, to redeploy each time, do:
 
-```
+```text
```
Comment (RaynalHugo) [inline strapi/README.md:23]:
```suggestion
```sh
```

File: strapi/README.md [hunk 5]
```diff
@@ -20,21 +20,21 @@ Follow this https://strapi.io/documentation/developer-docs/latest/setup-deployme
...
 git subtree push --prefix strapi heroku main
 ```
 
 To see production logs:
 
-```
+```text
```
Comment (RaynalHugo) [inline strapi/README.md:29]:
```suggestion
```sh
```

File: strapi/README.md [hunk 6]
```diff
@@ -20,21 +20,21 @@ Follow this https://strapi.io/documentation/developer-docs/latest/setup-deployme
...
 heroku logs --tail 
 ```
 
 ## How was this setup ?
 
 It was setup with:
 
-```
+```text
```
Comment (RaynalHugo) [inline strapi/README.md:37]:
```suggestion
```sh
```

File: .markdownlintignore (hunk 2 above)
Comment (bbenoist) [reply .markdownlintignore:3]: Nice catch!
Because, at the time of writing this commit (a few days before I started working), I was not aware that iog was one of our projects 😅 
In the doubt, I believed that iog was a copy of the source code from an external project so I avoided linting its files.

- [ ] Now that it's clearer to me, I'll remove this line and clean iog documentation 🚀

File: strapi/README.md (hunk 3 above)
Comment (bbenoist) [reply strapi/README.md:9]: @RaynalHugo The problem with using `sh` instead of `text` for declaring command-line stuff (not shell scripts) is that:

* it adds unwanted syntax coloring in some Markdown viewer

* it prevent use-cases where you want to display console output and/or prepend the line with a `$` or `#`. e.g.

```text
$ echo hello
hello
```

File: .eslintrc (hunk 1 above)
Comment (bbenoist) [reply .eslintrc:3]: Hmm, why not! They are interesting extras 👀 

I have opened a dedicated issue since I doubt that i'd be straightforward to enable: https://github.com/labelflow/labelflow/issues/661
